lib/ftp/bifrost.ex

```diff
@@ -454,6 +454,30 @@ defmodule Ftp.Bifrost do
     end
   end
 
+    # State, Path -> {ok, Size} OR {error, ErrorCause}
+    def size(connection_state() = conn_state, path) do
+      conn_state
+      |> unpack_state()
+      |> size(to_string(path))
+      |> pack_state(conn_state)
+    end
+  
+    def size(
+          %State{
+            permissions: permissions,
+            root_dir: root_dir,
+            current_directory: current_directory
+          } = state,
+          path
+        ) do
+      working_path = determine_path(root_dir, current_directory, path)
+  
+      case encode_file_info(permissions, working_path) do
+        nil -> {"-1", state}
```

Comment:
Instead of an esoteric size of -1 could we use something more descriptive, `:unknown` seems better to me.

Reply:
Or if we know the reason we could put that there. I also see in other plaves we just return `:error` and no state, I think.